### PR TITLE
[menu-bar] Setup react-native-web

### PR DESCRIPTION
**Pull request by gabrieldonadel**
# Why

This is the first of a series of PRs that intend to add experimental support for other platforms (Windows and Linux) using  Electron. 

# How

- Add react-native-web dependencies 
- Add mock components for native modules 
- Remove usage of the deprecated StyleSheet.compose API

# Test Plan

`npx expo start --web`

image

### CODE REVIEW

File: apps/menu-bar/macos/ExpoMenuBar-macOS/AppDelegate.m [hunk 1]
```diff
@@ -39,7 +39,7 @@ - (void)applicationDidFinishLaunching:(NSNotification *)aNotification {
   [statusItem.button setAction:@selector(onPressStatusItem:)];
 
   RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:_bridge
-                                                moduleName:@"ExpoMenuBar"
+                                                moduleName:@"main"
```
Comment (gabrieldonadel) [inline apps/menu-bar/macos/ExpoMenuBar-macOS/AppDelegate.m:42]: when using Expo `registerRootComponent` it sets the moduleName as `main`

File: apps/menu-bar/src/utils/create-component-primitive.tsx [hunk 2]
```diff
@@ -90,7 +90,7 @@ function stylesForVariants(props: any, variants: any = {}) {
 
       const styleValue = variants[key][value];
       if (styleValue) {
-        styles = StyleSheet.flatten(StyleSheet.compose(styles, styleValue));
+        styles = StyleSheet.flatten([styles, styleValue]);
```
Comment (gabrieldonadel) [inline apps/menu-bar/src/utils/create-component-primitive.tsx:93]: StyleSheet.compose has been deprecated on -react-native-web and throws a bunch of warnings

File: apps/menu-bar/index.js [hunk 3]
```diff
@@ -2,11 +2,13 @@
...
 
-import { name as appName } from './app.json';
 import App from './src/App';
 import './src/windows';
 
-AppRegistry.registerComponent(appName, () => App);
+// registerRootComponent calls AppRegistry.registerComponent('main', () => App);
+// It also ensures that whether you load the app in the Expo client or in a native build,
+// the environment is set up appropriately
+registerRootComponent(App);
```
Comment (gabrieldonadel) [inline apps/menu-bar/index.js:14]: We need to use expo `registerRootComponent` for react-native-web to work out of the box

File: apps/menu-bar/src/modules/PlatformColor/index.web.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export const PlatformColor = () => {
```
Comment (gabrieldonadel) [inline apps/menu-bar/src/modules/PlatformColor/index.web.ts:1]: Importing `PlatformColor` on react-native-web is not supported

File: apps/menu-bar/src/modules/Storage.ts [hunk 5]
```diff
@@ -61,15 +62,16 @@ export const resetStorage = () => {
   storage.clearAll();
 };
 
-const oldStorage = new MMKV();
 export const storage = new MMKV({
   id: StorageUtils.MMKVInstanceId,
-  path: StorageUtils.getExpoOrbitDirectory(MenuBarModule.homedir),
+  path:
```
Comment (gabrieldonadel) [inline apps/menu-bar/src/modules/Storage.ts:67]: MMKV does not support the path option on web

**Review by alanjhughes — APPROVED** [review]

**Pull request merged**